Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveHM_2nd.cpp

```diff
@@ -510,18 +538,23 @@ void FiniteDifferenceSolver::MacroscopicEvolveHMCartesian_2nd(
                         if (mag_exchange_coupling == 1){
                             // H_exchange
                             if (mag_exchange_arrx == 0._rt) amrex::Abort("The mag_exchange_arrx is 0.0 while including the exchange coupling term H_exchange for H_eff");
+
+                            LapM_prev_xface(i, j, k, 0) = T_Algo::Laplacian(M_prev_xface, coefs_x, coefs_y, coefs_z, i, j, k, 0); // M_xface from previous iteration of current time step
+                            LapM_prev_xface(i, j, k, 1) = T_Algo::Laplacian(M_prev_xface, coefs_x, coefs_y, coefs_z, i, j, k, 1); // M_xface from previous iteration of current time step
+                            LapM_prev_xface(i, j, k, 2) = T_Algo::Laplacian(M_prev_xface, coefs_x, coefs_y, coefs_z, i, j, k, 2); // M_xface from previous iteration of current time step
+
                             amrex::Real const H_exchange_coeff = 2.0 * mag_exchange_arrx / PhysConst::mu0 / mag_Ms_arrx / mag_Ms_arrx;
-                            Hx_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 0);
-                            Hy_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 1);
-                            Hz_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 2);
+                            Hx_eff += H_exchange_coeff * LapM_prev_xface(i, j, k, 0);
+                            Hy_eff += H_exchange_coeff * LapM_prev_xface(i, j, k, 1);
+                            Hz_eff += H_exchange_coeff * LapM_prev_xface(i, j, k, 2);
                         }
 
                         if (mag_anisotropy_coupling == 1){
                             // H_anisotropy
                             if (mag_anisotropy_arrx == 0._rt) amrex::Abort("The mag_anisotropy_arrx is 0.0 while including the anisotropy coupling term H_anisotropy for H_eff");
                             amrex::Real M_dot_anisotropy_axis = 0.0;
                             for (int comp=0; comp<3; ++comp) {
-                                M_dot_anisotropy_axis += M_xface(i, j, k, comp) * anisotropy_axis[comp];
+                                M_dot_anisotropy_axis += M_prev_xface(i, j, k, comp) * anisotropy_axis[comp];
```

Comment:
I noticed this little bug when fixing other parts. In our 2nd order algorithm, we need Heff at current time step from previous iteration, thus `M_prev` should be used here, instead of `M`